cmd/kubeadm/app/util/tokens_test.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+func TestUsingEmptyTokenFails(t *testing.T) {
+	// Simulates what happens when you omit --token on the CLI
+	s := newSecretsWithToken("")
+
+	ok, err := UseGivenTokenIfValid(s)
+	assert.False(t, ok)
+	assert.NoError(t, err)
```

Comment:
I'd love to discuss this more, but I think offline face-to-face might be more effective. Similar to coding style, this is the kind of thing that I think merits exactly one discussion per project, and then the consensus should just be documented as the project's standard. Otherwise, without a standard there will be no consistency in the codebase, and the same discussion will keep popping up and eating developer cycles whenever someone from camp A reviews code from someone in camp B and they have to rehash the same points. As much as I like unit test assertions, I far prefer consistency and putting the discussion to rest. :-)

I wonder if there's a happy medium, though, to create our own internal assertion helpers that address the points you mentioned. They can make failure descriptions required, have very concise output, and avoid the pitfalls of including anything like JSONEq, while still delivering more succinct checks than manual ifs in tests. Happy to discuss more.

For now, though, I'll ditch the assertion library and give vanilla Go testing a try. Thanks!